### PR TITLE
feat(indexing,mcp): byte-range provenance for chunks

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

Chunks now carry their exact `byte_start`/`byte_end` UTF-8 byte offsets, and the MCP context-assembly path surfaces those offsets in citation headers. Agents (and humans) can locate the precise region of a source file a chunk came from without re-running retrieval or scanning the whole file.

This is a follow-up to #22 (MCP server) and an instance of the cocoindex-inspired *lineage* pattern: every retrieved chunk traces back to the exact source byte range.

## Changes

- `DocumentProcessor.process_text` / `process_file` compute UTF-8 byte offsets per token and attach `byte_start` / `byte_end` to every chunk's metadata (single-chunk, multi-chunk, and error paths). `process_file` also records `source_bytes` so callers can sanity-check `byte_end <= source_bytes`.
- `Document.byte_start` / `byte_end` fields persist offsets at the Document layer; `get_chunk_info` returns a 4-tuple including them.
- `_format_byte_range` helper formats a ` (bytes N-M)` suffix from chunk metadata, with guards for missing / partial / non-int / inverted ranges. The MCP `_assemble_context` appends the byte range to the existing `*Source: ...*` citation line when present.

## Example MCP output

```
### [80% relevant] notes.md
*Source: /tmp/notes.md (bytes 1024-2048)*

Chunk body text.
```

Chunks indexed before this change are unaffected — when `byte_start`/`byte_end` are absent, the citation header looks exactly as it did before.

## Test plan

- [x] `pytest tests/test_document_processor.py` → 15 passed (incl. 4 new tests: source_bytes metadata, monotonic chunk offsets, single-chunk full-file ranges, UTF-8 multi-byte safety)
- [x] `pytest tests/test_chunking.py` → 7 passed (incl. 2 new end-to-end tests verifying offsets survive the indexer + ChromaDB roundtrip and stay within `source_bytes`)
- [x] `pytest tests/test_mcp_server.py` → 20 passed (incl. 4 new tests covering `_format_byte_range` edge cases plus `_assemble_context` byte-range surfacing)
- [x] Pre-commit (ruff / mypy) clean

Tracked as idea #213 in Bob's strategic backlog (gptme-rag byte-range provenance).

### CODE REVIEW

File: gptme_rag/indexing/document_processor.py [hunk 1]
```diff
@@ -32,42 +32,79 @@ def __init__(
...
+        """
+        offsets: list[int] = []
+        # Build cumulative byte-length lookups for decoded prefixes
+        decoded = ""
+        byte_pos = 0
+        for token in tokens:
+            offsets.append(byte_pos)
+            decoded += self.encoding.decode([token])
+            byte_pos = len(decoded.encode("utf-8"))
+        return offsets
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme_rag/indexing/document_processor.py:58]: Fixed in 5841dcc — now tracks a running byte total incrementally instead of re-encoding the full accumulated string on every iteration. O(1) per token instead of O(n).

File: gptme_rag/mcp_server.py [hunk 2]
```diff
@@ -14,6 +14,23 @@
...
+    region of the source file the chunk came from without re-running retrieval.
+    """
+    if not metadata:
+        return ""
+    start = metadata.get("byte_start")
+    end = metadata.get("byte_end")
+    if not isinstance(start, int) or not isinstance(end, int):
+        return ""
+    if end < start:
+        return ""
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme_rag/mcp_server.py:30]: Fixed in 118c51f — changed `end < start` to `end <= start` so a zero-byte range suppresses the suffix rather than emitting `(bytes N-N)`.
